### PR TITLE
Add clusterroles rbac to obs cluster and clean-up

### LINKED ISSUE #308: Provide Redhat contributors with admin access to nerc-ocp-obs cluster
A team of Redhat folks, lead by @computate will need to have admin rights to the new nerc-ocp-obs cluster.

It was proposed that a new team is created in OCP-on-NERC called nerc-obs-admin to control this, and following contributors added as members:

bnshr
jbasu01
schwesig
harshil-codes

### PR DESCRIPTION
**Pull request by schwesig**
Add clusterroles rbac to obs cluster and clean-up

- affects obs-cluster
- add clusterroles rbac
- streamline: remove redundant github-client-secret.yaml (follow https://github.com/OCP-on-NERC/nerc-ocp-config/pull/316)
- adjust kustomization.yaml to that
- cleanup: remove oauths-clientsecret-nerc, keycloak.mss not needed  (yet)
- adjust rook-ceph-external-cluster to fit to changes in https://github.com/OCP-on-NERC/nerc-ocp-config/pull/348

fixes https://github.com/nerc-project/operations/issues/308

Signed-off-by: ​/Thor(sten)?/ Schwesig <89909507+schwesig@users.noreply.github.com>

### CODE REVIEW

**Review by computate — APPROVED** [review]

File: cluster-scope/overlays/nerc-ocp-obs/kustomization.yaml [hunk 1]
```diff
@@ -53,13 +55,6 @@ patches:
             - ocp-on-nerc/nerc-ops
             - ocp-on-nerc/nerc-logs-metrics
             - ocp-on-nerc/nerc-obs-admins
-- target:
-    kind: ExternalSecret
-    name: oauths-clientsecret-nerc
-  patch: |
-    - op: replace
-      path: /spec/data/0/remoteRef/key
-      value: nerc/nerc-ocp-obs/openshift-config/oauths-clientsecret-nerc
```
Comment (larsks) [inline cluster-scope/overlays/nerc-ocp-obs/kustomization.yaml:65]: Why are we removing this patch?

Comment (schwesig) [reply cluster-scope/overlays/nerc-ocp-obs/kustomization.yaml:65]: @larsks I thought they are only for keycloak.mss (only in prod right now), and I assumed we don't want this (yet) for the obs, therefore I streamlined it.
But maybe I am wrong in one or both of these thoughts?

Comment (larsks) [reply cluster-scope/overlays/nerc-ocp-obs/kustomization.yaml:65]: Thanks, that makes sense.

Would you update the commit message to include this information?

**Review by jbasu01 — APPROVED** [review]
Looks good

File: cluster-scope/overlays/nerc-ocp-obs/kustomization.yaml (hunk 1 above)
Comment (schwesig) [reply cluster-scope/overlays/nerc-ocp-obs/kustomization.yaml:65]: thanks, will do.

**Comment by larsks** [comment]
In general, I prefer rebase over merge for incorporating changes from the main branch (it leads to a less confusing history)...but don't bother changing this pr unless you're already planning on making other changes (it's not that important).

**Comment by schwesig** [comment]
> In general, I prefer rebase over merge for incorporating changes from the main branch (it leads to a less confusing history)...but don't bother changing this pr unless you're already planning on making other changes (it's not that important).

@larsks ack 👍 I was a little lost in resolving the rook validation error to adjust to the in between merge of a PR. Will pay attention to it next time.

**Review by larsks — APPROVED** [review]

**Pull request merged**